src/lib/checkEngine.ts

```diff
@@ -9,3 +9,12 @@ const requiredVersion = getPkg().engines.node;
 if (!satisfies(process.version, requiredVersion)) {
     throw new Error(`Unsupported Node version. Expected ${ requiredVersion }, got ${ process.version }`);
 }
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const isNpmNotYarn = Boolean(process.env.npm_execpath) && !/\byarn\b/.test(process.env.npm_execpath!);
+
+if (isNpmNotYarn) {
```

Comment:
If you remove `Boolean` wrap, then non-null assertion isn't required anymore:
```suggestion
if (process.env.npm_execpath && !/\byarn\b/.test(process.env.npm_execpath)) {
```

Reply:
but then it will complain about `Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`

Reply:
so to bypass that, first, we need to check if it exists (to be safe) and then pass the value as `non-null` assertion.